ruby/ql/lib/codeql/ruby/frameworks/ActionController.qll

```diff
@@ -505,6 +506,27 @@ private module ParamsSummaries {
       ]
   }
 
+  /** Gets a field of an instance of `ActionController::Parameters` */
+  private DataFlow::LocalSourceNode paramsField() {
+    result =
+      [
+        paramsInstance(),
+        paramsInstance().getAMethodCall(methodReturnsTaintFromSelf()).getAnElementRead*()
```

Comment:
This is probably a bit broader than it needs to be, as it will include field accesses like `params[:foo]` and `params.dig(:foo, :bar, :baz)`, but also other things like hashes and arrays that are tainted by request parameters (e.g. `params`, `params.deep_dup()`, `params.values_at(:a, :b)`). 

This should work fine in practice, but the name and documentation is misleading - this is really describing anything that might be tainted by params in a way that our standard taint tracking might not detect.